bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -87,25 +103,82 @@ public FlatRow next() throws IOException {
       try {
         FlatRow result = currentDelegate.next();
         if (result != null) {
-          retryHandler.update(result);
+          requestManager.updateLastFoundKey(result.getRowKey());
+          // We've had at least one successful RPC, reset the backoff and retry counter
+          currentErrorBackoff = null;
+          timeoutRetryCount = null;
```

Comment:
Might be less error prone to initialize the AtomicInteger in the constructor & set it to 0 here